Break up the rcos.sql file

rushsteve1 opened the pull request:
This PR breaks up the massive `rcos.sql` and converts it and the other SQL files into [DbMate](https://github.com/amacneil/dbmate) migrations. Migrations are ordered according to the timestamp at the start of their filename. Ignore some odd sequential timestamps, some things needed manual re-ordering.

Along with this I took the time to refactor much of the SQL and to cleanup a lot of the weirdness created from generating it from DBML. Along with various other things like some `CHECK`s, a new checked `url` type, and improved relations.

These migrations are fully functional and can be applied with `dbmate up` and rolled all the way back down to a clean database with `dbmate down` (you have to run that repeatedly since it only goes on at a time).

rushsteve1 committed:
Slay the Beast

NOTE: This has NOT been tested yet, that is for the next commit!

Broke up the massive rcos.sql into a large number of smaller files that
conform to the format used by DbMate[1], which is similar to the format
used by other migration tools with an up and down statement.

While going through I made sure to cleanup the SQL, condensing, adding
comments, rewriting foreign key statements, and adding CHECKs, and
improving the formatting.

The new SQL format is hopefully easier to understand since everything is
broken into distinct related components with a clear up and down.

Next up is testing, additional checks, and more cleanup. Then on to more
views and stored functions.

[1]: https://github.com/amacneil/dbmate

rushsteve1 committed:
Fix all the bugs!

There were a number of typos, syntax errors, ordering problems, and
other such issues throughout the migrations. But they're all gone now!

The migrations can run through a full `dbmate up` and roll all the way
back down to a clean database without issue!

Apexal reviewed (approved):
Good call on change schema back to `public`. There was no real reason to make it `api`, I was just following the Postgrest doc example.

rushsteve1 committed:
Use URL type and split meeting location and URL

Used the URL type in a few missing places, and separated the concept of
the meeting's location and URL, with an is_remote toggle and a check to
enforce it.

rushsteve1 committed:
Semester ID format check

Added a simple regex check for semester IDs to ensure they're always 6
numbers.

rushsteve1 committed:
No ending period in comments

Normalized all the SQL comment declarations to not have periods at the
end of them.
